### PR TITLE
Ux improvement

### PR DESCRIPTION
**Pull request by Rajesh-Khanna**

### CODE REVIEW

File: src/apps/sketch/components/Board/sketchBoard.jsx [hunk 1]
```diff
@@ -15,31 +19,62 @@ export default function SketchBoard(props) {
...
+                    paletteHandler.color('eraser');
+                    break;
+                case '+':
+                    paletteHandler.increaseFont();
+                    break;
+                case '-':
+                    paletteHandler.reduceFont();
+                    break;
+                default:
+                    console.log('');
```
Comment (theVirtualMan) [inline src/apps/sketch/components/Board/sketchBoard.jsx:44]: why empty string?

File: src/apps/sketch/components/Board/sketchBoard.jsx [hunk 2]
```diff
@@ -15,31 +19,62 @@ export default function SketchBoard(props) {
...
             currShape.setFont(props.font);
             currShape.setColor(props.color);
             currShape.setStart(p.mouseX, p.mouseY);
             currShape.setEnd(p.pmouseX, p.pmouseY);
-            const copyObj = new Shape();
-            copyObj.copy(currShape.clone());
-            if(brush)
-                brush.send(currShape.export());
-            currShape.drawShape(p);
+            if(!currShape.isStationary()){
```
Comment (theVirtualMan) [inline src/apps/sketch/components/Board/sketchBoard.jsx:66]: what does it do?

Comment (theVirtualMan) [reply src/apps/sketch/components/Board/sketchBoard.jsx:66]: what about  the case when user loops the diagram?

File: src/apps/sketch/components/Board/board.jsx [hunk 3]
```diff
@@ -26,14 +28,34 @@ const Board = props => {
...
   return (
     <Row>
       <Col xs={24} xl={20}>
         {
           brush 
             ? <>
-                <SketchBoard brush = {brush} font = {font} color = {color}/>
-                <Buttons handleFont={handleFont} handleColor={handleColor}/>
+                <SketchBoard brush = {brush} font = {font} color = {color} paletteHandler = {paletteHandler} />
+                <Palette handleFont={handleFont} handleColor={handleColor}/>
```
Comment (theVirtualMan) [inline src/apps/sketch/components/Board/board.jsx]: better to change the button file name also, I see that it is defined in buttons.jsx file

File: package.json [hunk 4]
```diff
@@ -15,6 +15,7 @@
         "react-p5": "^1.3.6",
         "react-p5-wrapper": "^2.3.0",
         "react-scripts": "^4.0.1",
+        "react-window": "^1.8.6",
```
Comment (theVirtualMan) [inline package.json:18]: what is it required for?

File: src/apps/sketch/components/Board/sketchBoard.jsx (hunk 2 above)
Comment (Rajesh-Khanna) [reply src/apps/sketch/components/Board/sketchBoard.jsx:66]: our free drawing is a bunch of small straight lines so if u r forming a loop then the start point will be in one line and end point will be in another, 
reason for adding this is, now that we added a shortcut to draw by just clicking on d once, but if he is not moving the cursor the same data will be transerferd mulitple times a sec

File: src/apps/sketch/components/Board/sketchBoard.jsx (hunk 1 above)
Comment (Rajesh-Khanna) [reply src/apps/sketch/components/Board/sketchBoard.jsx:44]: just a placeholder

File: package.json (hunk 4 above)
Comment (Rajesh-Khanna) [reply package.json:18]: in react for us to add a message to the list we need to rerender the entire list, which will slow down the system, if the num of message are in 1000s this lib will only rerender the text that is visible on the screen

File: src/apps/sketch/components/Board/board.jsx (hunk 3 above)
Comment (Rajesh-Khanna) [reply src/apps/sketch/components/Board/board.jsx]: correcting

**Pull request merged**